presto-main/src/test/java/com/facebook/presto/operator/scalar/TestRegexpFunctions.java

```diff
@@ -141,6 +141,22 @@ public void testRegexpReplace()
         assertFunction("REGEXP_REPLACE(VARCHAR 'x', '.*', 'xxxxx')", createUnboundedVarcharType(), "xxxxxxxxxx");
     }
 
+    @Test
+    public void testRegexpReplaceLambda()
+    {
+        assertFunction("REGEXP_REPLACE('new', '(\\w)', x -> x[1])", createUnboundedVarcharType(), "new");
+        assertFunction("REGEXP_REPLACE('new york', '(\\w)(\\w*)', x -> upper(x[2]) || lower(x[3]))", createUnboundedVarcharType(), "New York");
+        assertFunction("REGEXP_REPLACE('new', '(\\w*)', x -> upper(x[1]))", createUnboundedVarcharType(), "NEW");
+        assertFunction("REGEXP_REPLACE('new', '(\\w*)', x -> cast(length(x[1]) AS varchar))", createUnboundedVarcharType(), "30");
+        assertFunction("REGEXP_REPLACE('new york', '(n\\w*)', x -> upper(x[1]))", createUnboundedVarcharType(), "NEW york");
+        assertFunction("REGEXP_REPLACE('new york', '(y\\w*)', x -> upper(x[1]))", createUnboundedVarcharType(), "new YORK");
+        assertFunction("REGEXP_REPLACE('new york city', '(yo\\w*)', x -> upper(x[1]))", createUnboundedVarcharType(), "new YORK city");
+        assertFunction("REGEXP_REPLACE('xxx xxx xxx', 'x', x -> upper(x[1]))", createUnboundedVarcharType(), "XXX XXX XXX");
+        assertFunction("REGEXP_REPLACE('123 456', '([0-9]*)', x -> x[1])", createUnboundedVarcharType(), "123 456");
+        assertFunction("REGEXP_REPLACE('123 456', '(([0-9]*) ([0-9]*))', x -> x[3] || x[4])", createUnboundedVarcharType(), "123456");
+        assertInvalidFunction("REGEXP_REPLACE('new', '(\\w*)', x -> upper(x[3]))", INVALID_FUNCTION_ARGUMENT);
```

Comment:
add tests with empty matches.

Reply:
Also add tests for

- cases where there is no match: `regexp_replace('something', '(a)', ...)`
- empty input: `regexp_replace('', '(a)', ...)`
- null input: `regexp_replace(null, '(a)', ...)`
- null pattern: `regexp_replace('something', null, ...)`
- overlapping matches: `regexp_replace('abbabba', 'abba', x -> 'x') → 'xbba'`
- where there is match but a group is not matched: `REGEXP_REPLACE('abde', 'ab(c)?de', x -> coalesce(x[1], 'OK'))` (expected `'OK'`, assuming lack of group-match is represented as `null` entry in lambda input array)

Reply:
Another, distinct, overlapping matches case: `regexp_replace('abbabba', 'abba', x -> 'm' || x[1]) → 'mabbabba'`

Reply:
@sopel39 If you were talking about the updating Joni match start by 1 for empty match case, there was a case related to that: 
REGEXP_REPLACE('new', '(\\w*)', x -> cast(length(x[1]) AS varchar))" = "30"
For this pattern there are 2 matches [0,3] and [3,3] and the second is empty match. The length for it is 0. I also added the following test in the new iteration:

REGEXP_REPLACE('new york', '(\\w*)', x -> '<' || x[1] || '>') = "<new><> <york><>"

Reply:
I meant some test like:
```
REGEXP_REPLACE('new', '', x -> 'X') = "XnXeXwX"
```
and maybe also:
```
REGEXP_REPLACE('new', '.', x -> 'X') = "XXX"
```